src/GraphWalker.cpp

```diff
@@ -39,7 +39,7 @@ static bool validConnection(Structure* src, Structure* dst, Direction _d)
 	if (src == nullptr || dst == nullptr)
 	{
 		#ifdef DEBUG
-		throw std::runtime_error("GraphWalker::validConnection() was passed a NULL Pointer.");
+		std::cout << "GraphWalker::validConnection() was passed a NULL Pointer." << std::endl;

```

Comment:
I'm not so fond of this. I think an exception would be better here.

I'm also not so fond of the surrounding `#ifdef`. I think that's the real source of the problem.

Reply:
It is. But I didn't write it, and at the time I was concerned with just getting rid of the warning. :wink:

I disagree on the exception. It's not a fatal unrecoverable state that a `nullptr` is passed in; it's just another case where the connection is not valid and `false` should be returned.

Reply:
Well, technically, exceptions are not unrecoverable. You can catch them. But that's beside the point here.

----

Being able to check for `nullptr` and doing nothing doesn't mean there's no problem. It could indicate the calling code is doing something stupid and needs to know about it. In this case, it really doesn't make sense to try and check for a path between two structures if you're not passing two valid structures.

If this ever happens, it's quite clearly a programming error, and the code should complain loudly. An exception is much better than a relatively silent log message, which could easily be missed.